File: tdd-kit/tdd-kit-ruby/lib/fizzbuzz_test.rb
require_relative './test_helper'
require_relative './fizzbuzz'

describe Fizzbuzz do
  before do
    @elements = Fizzbuzz.new.elements
  end

  it 'should return 100 elements' do
    @elements.length.must_equal 100
  end

  it 'should return "1" for 1' do
    fizzbuzz(1).must_equal '1'
  end

  it 'should return "fizz" for 3' do
    fizzbuzz(3).must_equal 'fizz'
  end

  it 'should return "buzz" for 5' do
    fizzbuzz(5).must_equal 'buzz'
  end

  it 'should return "fizzbuzz" for 15' do
    fizzbuzz(15).must_equal 'fizzbuzz'
  end

  private

  def fizzbuzz(number)
    @elements[number - 1]
  end
end

File: tdd-kit/tdd-kit-java/src/test/java/FizzbuzzTest.java
import org.junit.Before;
import org.junit.Test;

import java.util.List;

import static org.assertj.core.api.Assertions.*;

public class FizzbuzzTest {

    @Before
    public void before() throws Exception {
        elements = new Fizzbuzz().elements();
    }

    @Test
    public void should_return_100_elements() throws Exception {
        assertThat(elements.size()).isEqualTo(100);
    }

    @Test
    public void should_return_1_for_1() throws Exception {
        assertThat(fizzBuzzOf(1)).isEqualTo("1");
    }

    @Test
    public void should_return_fizz_for_3() throws Exception {
        assertThat(fizzBuzzOf(3)).isEqualTo("fizz");
    }

    @Test
    public void should_return_buzz_for_5() throws Exception {
        assertThat(fizzBuzzOf(5)).isEqualTo("buzz");
    }

    @Test
    public void should_return_fizzbuzz_for_15() throws Exception {
        assertThat(fizzBuzzOf(15)).isEqualTo("fizzbuzz");
    }

    private String fizzBuzzOf(int number) {
        return elements.get(number - 1);
    }

    private List<String> elements;
}
